app/js/services/QueryBuilderService.js

```diff
@@ -36,90 +36,105 @@ function QueryBuilderService( $resource, $http, AppSettings, NotificationService
-        // Check to make sure we have at least 1 gene and 1 disease
-        if(diseases.length > 0 && genes.length > 0) {
-          // parse diseases for just their acronyms
-          diseases = diseases.map(diseaseObj =>diseaseObj['acronym']);
-          // parse genes for just their entrezgene
-          genes = genes.map(gene=>gene['entrezgene']);
-
-          // Submit our query
-          queryResource
-            .submit({diseases, genes})
-            .$promise.then(res=>{
+        // parse diseases for just their acronyms
+        diseases = diseases.map(diseaseObj =>diseaseObj['acronym']);
+        // parse genes for just their entrezgene
+        genes = genes.map(gene=>gene['entrezgene']);
 
-            NotificationService.notify({
-              type:    'success',
-              message: `<span class="glyphicon glyphicon-ok" aria-hidden="true"></span> Classifier #${res.id} submitted!  
-                  <div class="btn-group" style="float:right;" role="group" >
-                      <button type="button" class="btn btn-default">Review Submission</button>
-                      <button type="button" class="btn btn-default">View My Classifiers</button>
-                      <button type="button" class="btn btn-default">Create New Classifier</button>
-                  </div>`,
-              config:{ ttl: -1  } // stay open until users closes
+        // Submit our query
+        queryResource
+          .submit({diseases, genes})
+          .$promise.then(res=>{
 
-            });
+          NotificationService.notify({
+            type:    'success',
+            message: `<span class="glyphicon glyphicon-ok" aria-hidden="true"></span> Classifier #${res.id} submitted!
+                <div class="btn-group" style="float:right;" role="group" >
+                    <button type="button" class="btn btn-default">Review Submission</button>
+                    <button type="button" class="btn btn-default">View My Classifiers</button>
+                    <button type="button" class="btn btn-default">Create New Classifier</button>
+                </div>`,
+            config:{ ttl: -1  } // stay open until users closes
 
-          }, error => {
-            console.log(error);
-            NotificationService.notify({
-              type:    'error',
-              message: `Failed to submit classifier. Error: ${error}`
-            });
           });
 
-        } else {
+        }, error => {
+          console.log(error);
           NotificationService.notify({
             type:    'error',
-            message: 'You must select at least 1 disease and 1 gene.'
+            message: `Failed to submit classifier. Error: ${error}`
           });
-        }
+        });
+
       }
 
-      if (user.email === null || user.email === undefined || IGNORE_EXISTING_EMAIL) {
-        // update email
-        let email = prompt("Enter your email address to receive your classifier:")
+      function getEmail(){
```

Comment:
This function not only gets email but also calls `submitClassifier()`. You may want to rename it to something like `doSubmission`.

Reply:
I renamed it to `executeSubmission`. It's not a great name but much better than `getEmail`.